gbpservice/nfp/configurator/agents/vpn.py

```diff
@@ -0,0 +1,401 @@
+        """Update local status.
+
+        This method call updates status attribute of
+        VPNServices.
+        """
+        msg = {'receiver': const.NEUTRON,
+               'resource': const.SERVICE_TYPE,
+               'method': 'update_status',
+               'data': {'context': context,
+                        'status': status}
+               }
+        self.notify._notification(msg)
+
+    def ipsec_site_conn_deleted(self, context, resource_id):
+        """ Notify VPNaaS plugin about delete of ipsec-site-conn """
+
+        msg = {'receiver': const.NEUTRON,
+               'resource': const.SERVICE_TYPE,
+               'method': 'ipsec_site_conn_deleted',
+               'data': {'context': context,
+                        'resource_id': resource_id}
+               }
+        self.notify._notification(msg)
+
+""" Implements VPNaasRpcManager class which receives requests
+    from Configurator to Agent.
+
+Methods of this class are invoked by the configurator. Events are
+created according to the requests received and enqueued to worker queues.
+
+"""
+
+
+class VPNaasRpcManager(agent_base.AgentBaseRPCManager):
+
+    RPC_API_VERSION = '1.0'
+    target = messaging.Target(version=RPC_API_VERSION)
+
+    def __init__(self, conf, sc):
+        """Instantiates child and parent class objects.
+
+        Passes the instances of core service controller and oslo configuration
+        to parent instance in order to provide event enqueue facility for batch
+        processing event.
+
+        :param sc: Service Controller object that is used for interfacing
+        with core service controller.
+        :param conf: Configuration object that is used for configuration
+        parameter access.
+
+        """
+
+        super(VPNaasRpcManager, self).__init__(conf, sc)
+
+    def vpnservice_updated(self, context, kwargs):
+        """Registers the VPNaas plugin events to update the vpn configurations.
+
+        :param context: dictionary, confined to the specific service type.
+        :param kwargs: dictionary, confined to the specific operation type.
+
+        Returns: None
+        """
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.new_event(id='VPNSERVICE_UPDATED', data=arg_dict)
+        self._sc.post_event(ev)
+
+"""
+Handler class to invoke the vpn driver methods.
+For every event that gets invoked from worker process lands over here
+to make a call to the driver methods.
+"""
+
+
+class VPNaasEventHandler(core_pt.PollEventDesc):
+
+    def __init__(self, sc, drivers):
+        """ Instantiates class object.
+
+        :param sc: Service Controller object that is used to communicate
+        with process model core file.
+        :param drivers: dictionary of driver name to object mapping
+
+        """
+        self._sc = sc
+        self._drivers = drivers
+        self._plugin_rpc = VpnaasRpcSender(self._sc)
+
+    def _get_driver(self):
```

Comment:
This only supports one driver. 
If there are different drivers like vyos and asav, then we need to use some identifier like "service_type"